src/crimson/os/seastore/transaction.h

```diff
@@ -84,14 +84,20 @@ class Transaction {
     ceph_assert(inserted);
   }
 
-  void add_fresh_extent(CachedExtentRef ref) {
+  void add_fresh_extent(CachedExtentRef ref, bool set_addr = true) {
     ceph_assert(!is_weak());
     fresh_block_list.push_back(ref);
-    ref->set_paddr(make_record_relative_paddr(offset));
-    offset += ref->get_length();
+    if (set_addr) {
+      ref->set_paddr(make_record_relative_paddr(offset));
+      offset += ref->get_length();
+    }
     write_set.insert(*ref);
```

Comment:
`write_set` is uniqually indexed by paddrs, so I don't think we can insert extent with the same null address. And we need to lookup these extents in write_set by their paddrs.

Reply:
We are not inserting extents with same null addresses, we insert extents with "delayed_temp_paddr" which is unique to each extent.

Reply:
Oh, it's in a later commit, still looking.